crates/goose/src/providers/ollama.rs

```diff
@@ -51,9 +53,28 @@ impl OllamaProvider {
     }
 
     async fn post(&self, payload: Value) -> Result<Value, ProviderError> {
-        let url = format!("{}/v1/chat/completions", self.host.trim_end_matches('/'));
+        // OLLAMA_HOST is sometimes just the 'host' or 'host:port' without a scheme
+        let base = if self.host.starts_with("http://") || self.host.starts_with("https://") {
+            self.host.clone()
+        } else {
+            format!("http://{}", self.host)
+        };
+
+        let mut base_url = Url::parse(&base)
+            .map_err(|e| ProviderError::RequestFailed(format!("Invalid base URL: {e}")))?;
+
+        // Set the default port if missing
+        if base_url.port().is_none() {
```

Comment:
if someone were running ollama on http/80 or https/443, they wouldn't need the port, and this would just add `OLLAMA_DEFAULT_PORT`, we should make it clear that if no port is set it will use `OLLAMA_DEFAULT_PORT` and let the user know if they're running on 80 or 443 to always include it